src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -159,28 +136,75 @@ private static List<CompletionResult> CompleteCommand(CompletionContext context,
                     moduleName = commandName.Substring(0, indexOfFirstBackslash);
                     commandName = commandName.Substring(indexOfFirstBackslash + 1);
 
-                    var powershell = context.Helper
-                        .AddCommandWithPreferenceSetting("Get-Command", typeof(GetCommandCommand))
-                        .AddParameter("All")
-                        .AddParameter("Name", commandName)
-                        .AddParameter("Module", moduleName);
+                    commandResults = ExecuteGetCommandCommand(context, moduleName, types, includeModulePrefix: true);
+                }
+            }
 
-                    if (!types.Equals(CommandTypes.All))
-                        powershell.AddParameter("CommandType", types);
+            return commandResults;
+        }
 
-                    Exception exceptionThrown;
-                    var commandInfos = context.Helper.ExecuteCurrentPowerShell(out exceptionThrown);
+        private static List<CompletionResult> ExecuteGetCommandCommand(CompletionContext context, string moduleName, CommandTypes types, bool includeModulePrefix)
```

Comment:
I recommend to have this method as a local function within `CompleteCommand`. This method recalculate `addAmpersandIfNecessary` and `quote`. We can avoid the duplication if we use a local function.

Reply:
will do